packages/visualizations-react/.eslintrc.js

```diff
@@ -1,8 +1,107 @@
 module.exports = {
-    extends: ['react-app', 'prettier/@typescript-eslint', 'plugin:prettier/recommended'],
+    env: {
+        browser: true,
+        es2021: true,
+    },
+    extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/recommended',
+        "plugin:react/recommended",
+        'airbnb',
+        'airbnb/hooks',
+        'airbnb-typescript',
+        'prettier',
+    ],
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: './tsconfig.json',
+        tsconfigRootDir: __dirname,
+        ecmaFeatures: {
+            "jsx": true
+        }
+    },
     settings: {
         react: {
             version: 'detect',
         },
     },
+    plugins: ['@typescript-eslint', 'prettier', 'react', 'no-only-tests'],
+    globals: {
+        'jest': true
+    },
+    rules: {
+        semi: ['error', 'always'],
+        '@typescript-eslint/semi': ['error', 'always'],
+        'max-len': [
+            'error',
+            {
+                code: 120,
+                tabWidth: 4,
+                ignoreStrings: true,
+                ignoreTemplateLiterals: true,
+                ignoreComments: true,
+            },
+        ],
+        camelcase: 0,
+        'no-plusplus': 0,
+        // disallow reassignment of function parameters
+        // disallow parameter object manipulation except for specific exclusions
+        'no-param-reassign': [
+            'error',
+            {
+                props: true,
+                ignorePropertyModificationsFor: [
+                    'acc', // for reduce accumulators
+                    'e', // for e.returnvalue
+                ],
+            },
+        ],
+        'object-curly-newline': [
+            'error',
+            {
+                ObjectExpression: {
+                    minProperties: 6,
+                    multiline: true,
+                    consistent: true,
+                },
+                ObjectPattern: {
+                    minProperties: 6,
+                    multiline: true,
+                    consistent: true,
+                },
+            },
+        ],
+        'import/no-named-as-default': 'off',
+        'import/no-extraneous-dependencies': [
+            'error',
+            { devDependencies: ['./test/**', '/stories/**'] },
+        ],
+        'no-underscore-dangle': 0,
+        'react/jsx-one-expression-per-line': 0,
+        'arrow-parens': [2, 'as-needed', { requireForBlockBody: true }],
+        'react/jsx-props-no-spreading': 0,
+        'jsx-a11y/label-has-for': 0,
+        'jsx-a11y/label-has-associated-control': ['error', {
+            labelComponents: ['label'],
+            controlComponents: ['input', 'CheckboxInput', 'RadioInput', 'SelectInput', 'TextInput', 'SwitchInput'],
+            depth: 25,
+        }],
```

Comment:
This rule seems to be very related to the platform code, I wonder if it is relevant to define it here. It also names components that are only present in the platform. Such as `CheckboxInput`, `RadioInput`, `SelectInput`, `TextInput` and `SwitchInput`.

Reply:
Yup, pure copy-paste, I'll have to rework that indeed. I'm not sure we need it at all, a11y should be done in `visualization` mainly I guess.

Reply:
Agreed, removed here 766042c1e7a4e7d0d0096a7f52bc737611226797
Note that if we release filters somewhere, we may want a similar rule again to make sure we don't have a text input without a label for example